apollo-runtime/src/main/java/com/apollographql/android/cache/HttpCache.java

```diff
@@ -51,12 +59,43 @@ boolean isStale(@Nonnull Response response) {
     return evictionStrategy.isStale(response);
   }
 
-  Response cacheProxy(@Nonnull Response response, @Nonnull String cacheKey) throws IOException {
-    ResponseCacheRecordEditor cacheRecordEditor = cacheStore.cacheRecordEditor(cacheKey);
-    new ResponseHeaderRecord(response).writeTo(cacheRecordEditor);
-    return response.newBuilder()
-        .body(new ResponseBodyProxy(cacheRecordEditor, response))
-        .build();
+  Response cacheProxy(@Nonnull Response response, @Nonnull String cacheKey) {
+    ResponseCacheRecordEditor cacheRecordEditor = null;
+    try {
+      cacheRecordEditor = cacheStore.cacheRecordEditor(cacheKey);
+      if (cacheRecordEditor != null) {
+        new ResponseHeaderRecord(response).writeTo(cacheRecordEditor);
+        return response.newBuilder()
+            .body(new ResponseBodyProxy(cacheRecordEditor, response))
+            .build();
+      }
+    } catch (Exception ignore) {
+      //TODO log me
+      abortQuietly(cacheRecordEditor);
+    }
+    return response;
+  }
+
+  private static void closeQuietly(ResponseCacheRecord cacheRecord) {
+    try {
+      if (cacheRecord != null) {
+        cacheRecord.close();
+      }
+    } catch (RuntimeException rethrown) {
+      throw rethrown;
+    } catch (Exception ignore) {
+    }
+  }
+
+  private static void abortQuietly(ResponseCacheRecordEditor cacheRecordEditor) {
+    try {
+      if (cacheRecordEditor != null) {
+        cacheRecordEditor.abort();
+      }
+    } catch (RuntimeException rethrown) {
```

Comment:
Would it be better to modify the method signature to throw that instead since it's always thrown when abort is called?